File: 0075.SortColors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = front = 0
        back = len(nums) - 1
        while i<=back:
            if nums[i] == 0:
                if i > front:
                    nums[front], nums[i] = nums[i], nums[front]
                i += 1
                front += 1
            elif nums[i] == 2:
                nums[back], nums[i] = nums[i], nums[back]
                back -= 1
            else:
                i += 1

File: 0000-0999/0075.SortColors.go
func sortColors(nums []int)  {
    l, r := 0, len(nums)-1
    for i := 0; i <= r; {
        if nums[i] == 0 {
            nums[i], nums[l] = nums[l], nums[i]
            l++
            i++
        } else if nums[i] == 2 {
            nums[i], nums[r] = nums[r], nums[i]
            r--
        } else {
            i++
        }
    }
    /*
    for i := 0; i <= r; i++ {
        for i < r && nums[i] == 2 {
            nums[i], nums[r] = nums[r], nums[i]
            r--
        }
        for i > l && nums[i] == 0 {
            nums[i], nums[l] = nums[l], nums[i]
            l++
        }
    }*/
}